planner/core/logical_plan_builder.go

```diff
@@ -6889,6 +6893,8 @@ func getResultCTESchema(seedSchema *expression.Schema, svar *variable.SessionVar
 		col.RetType = col.RetType.Clone()
 		col.UniqueID = svar.AllocPlanColumnID()
 		col.RetType.DelFlag(mysql.NotNullFlag)
+		// Since you have reallocated unique id here, the old-cloned-cached hash code is not valid anymore.
+		col.CleanHashCode()
```

Comment:
getResultCTESchema is called in 3 places, can you add unit tests to cover these paths?

Reply:
yes, it's ok. just a cache clean, hashcode still can be accessed with recomputation when using